File: CyberpunkTcgVault.Api/Models/WishListItem.cs
namespace CyberpunkTcgVault.Api.Models
{
    public class WishListItem
    {
        public int Id { get; set; }

        public int CardPrintingId { get; set; }

        public CardPrinting CardPrinting { get; set; } = null!;

        // Foreign key to the user who owns this wishlist item.
        public Guid UserId { get; set; }

        // Navigation property back to the owning user.
        public AppUser User { get; set; } = null!;

        public int WantedQuantity { get; set; }

        public string? Priority { get; set; }

        public string? ReasonWanted { get; set; }

        public bool WantRaw { get; set; }

        public bool WantGraded { get; set; }

        public string? PreferredGradingCompany { get; set; }

        public bool IsOpenToTrade { get; set; }

        public string? Notes { get; set; }
    }
}

File: CyberpunkTcgVault.Web/src/app/features/wishlist/models/wishlist-item.ts
/** Private wishlist record for one exact physical CardPrinting. */
export interface WishlistItem {
  id: number;
  cardPrintingId: number;
  cardId: number;
  cardName: string;
  setName: string | null;
  cardNumber: string | null;
  rarity: string | null;
  colour: string | null;
  imageUrl: string | null;
  wantedQuantity: number;
  priority: string | null;
  reasonWanted: string | null;
  wantRaw: boolean;
  wantGraded: boolean;
  preferredGradingCompany: string | null;
  isOpenToTrade: boolean;
  notes: string | null;
}

export interface CreateWishlistItemRequest {
  cardPrintingId: number;
  wantedQuantity: number;
}
